src/Sepes.Infrastructure/Service/Dataset/SandboxDatasetService.cs

```diff
@@ -59,36 +74,40 @@ public async Task<SandboxDatasetDto> Add(int sandboxId, int datasetId)
             if (datasetFromDb.StudyId.HasValue && datasetFromDb.StudyId != studyFromDb.Id)
             {
                 throw new ArgumentException($"Dataset {datasetId} cannot be added to Sandbox {sandboxId}. The dataset is Study specific and belongs to another Study than {studyFromDb.Id}.");
-            }  
+            }
 
-            // Create new linking table
+            var sandboxDatasetRelation = await _db.SandboxDatasets.FirstOrDefaultAsync(ds => ds.SandboxId == sandboxId && ds.DatasetId == datasetId);
+
+            //Is dataset allready linked to this sandbox?
+            if (sandboxDatasetRelation != null)
+            {
+                throw new ArgumentException($"Dataset is allready added to Sandbox.");
+            }
+
+            // Create new entry in the relation table
             var sandboxDataset = new SandboxDataset { SandboxId = sandboxId, DatasetId = datasetId, Added = DateTime.UtcNow, AddedBy = (await _userService.GetCurrentUserAsync()).UserName };
             await _db.SandboxDatasets.AddAsync(sandboxDataset);
             await _db.SaveChangesAsync();
+        }
 
-            return _mapper.Map<SandboxDatasetDto>(sandboxDataset);
-        }     
-
-        public async Task<SandboxDatasetDto> Remove(int sandboxId, int datasetId)
+        public async Task Remove(int sandboxId, int datasetId)
         {
             var studyFromDb = await StudySingularQueries.GetStudyBySandboxIdCheckAccessOrThrow(_db, _userService, sandboxId, UserOperation.Study_Crud_Sandbox);
```

Comment:
Brukes denne til noe?

Reply:
Det er den front end kaller for å fjerne dataset, altså når du "unchecker"